src/ReaderScreen/index.jsx

```diff
@@ -210,14 +201,14 @@ const Reader = ({ navigation, route }) => {
 
   const handleLoadEnd = useCallback(() => {
     // Scroll to saved element ID after WebView is fully loaded
-    if (webViewRef.current && currentElementId) {
+    if (webViewRef.current && currentElementIdRef.current) {
       const scrollMessage = {
         action: "scrollToPosition",
-        elementId: currentElementId,
+        elementId: currentElementIdRef.current,
       };
       webViewRef.current.postMessage(JSON.stringify(scrollMessage));
     }
-  }, [currentElementId]);
+  }, [currentElementIdRef.current]);
```

Comment:
Same here. 
It won't trigger re-render, so this callback won't trigger when currentElementIdRef changes